### ISSUE TITLE
False positive in tsan when locking more than 64 mutexes at once

### ISSUE THREAD

**Issue by kiroma** [llvm/llvm-project#57007]
[https://godbolt.org/z/3MbEz4eqY](https://godbolt.org/z/3MbEz4eqY)
```cpp
#include <mutex>

int main()
{
	std::mutex mtxs[65];
	for(int i = 0; i < 65; ++i)
	{
		mtxs[i].lock();
	}
	for(int i = 0; i < 65; ++i)
	{
		mtxs[i].unlock();
	}
}
```
This code reports the following and then deadlocks:
```
ThreadSanitizer: CHECK failed: sanitizer_deadlock_detector.h:67 "((n_all_locks_)) < (((sizeof(all_locks_with_contexts_)/sizeof((all_locks_with_contexts_)[0]))))" (0x40, 0x40) (tid=1)
```
